Proyecto Individual N°2 - Henry - Oliva Pinto

ETL - EXtraccion, Transformacion, Carga

El siguiente analisis se efectua con el objetivo de preparar y acondicionar la base de datos, para luego porder llevar a cabo anlisis mas profundos que permiten brindar informacion importante para mejorar la toma de decisiones.

Es buena practica observar tanto las primeras como las ultimas filas del archivo para tener un panorama genral del Data Frame, y ademas, identificar posibles anomalias. 

In [1]:
#Importamos las librerias necesarias
import pandas as pd
import numpy as np

In [2]:
#Cargamos el archivo como un Data Frame y visualizamos
aviones_db = pd.read_csv("/Users/tobiasolivapinto/Downloads/AccidentesAviones.csv")
#La funcion head nos permite ver las primeras filas
aviones_db.head()

Unnamed: 0               fecha HORA declarada  \
0           0  September 17, 1908           1718   
1           1  September 07, 1909              ?   
2           2       July 12, 1912           0630   
3           3     August 06, 1913              ?   
4           4  September 09, 1913           1830   

                                 Ruta                OperadOR flight_no  \
0                 Fort Myer, Virginia    Military - U.S. Army         ?   
1             Juvisy-sur-Orge, France                       ?         ?   
2           Atlantic City, New Jersey    Military - U.S. Navy         ?   
3  Victoria, British Columbia, Canada                 Private         ?   
4                  Over the North Sea  Military - German Navy         ?   

           route                 ac_type registration cn_ln all_aboard  \
0  Demonstration        Wright Flyer III            ?     1          2   
1       Air show          Wright Byplane          SC1     ?          1   
2    Test flight               Dirigible            ?     ?          5   
3              ?        Curtiss seaplane            ?     ?          1   
4              ?  Zeppelin L-1 (airship)            ?     ?         20   

  PASAJEROS A BORDO crew_aboard cantidad de fallecidos passenger_fatalities  \
0                 1           1                      1                    1   
1                 0           1                      1                    0   
2                 0           5                      5                    0   
3                 0           1                      1                    0   
4                 ?           ?                     14                    ?   

  crew_fatalities ground                                            summary  
0               0      0  During a demonstration flight, a U.S. Army fly...  
1               0      0  Eugene Lefebvre was the first pilot to ever be...  
2               5      0  First U.S. dirigible Akron exploded just offsh...  
3               1      0  The first fatal airplane accident in Canada oc...  
4               ?      0  The airship flew into a thunderstorm and encou...

In [3]:
#Visualizamos las ultimas filas del archivo
aviones_db.tail()

Unnamed: 0           fecha HORA declarada                        Ruta  \
5003        5003  March 28, 2021           1835          Near Butte, Alaska   
5004        5004    May 21, 2021           1800        Near Kaduna, Nigeria   
5005        5005   June 10, 2021           0800  Near Pyin Oo Lwin, Myanmar   
5006        5006   July 04, 2021          11:30  Patikul, Sulu, Philippines   
5007        5007   July 06, 2021           1500              Palana, Russia   

                             OperadOR flight_no                         route  \
5003                Soloy Helicopters         ?           Sightseeing Charter   
5004    Military - Nigerian Air Force         ?                             ?   
5005     Military - Myanmar Air Force         ?          Naypyidaw - Anisakan   
5006  Military - Philippine Air Force         ?  Cagayan de Oro-Lumbia - Jolo   
5007    Kamchatka Aviation Enterprise       251        Petropavlovsk - Palana   

                            ac_type registration   cn_ln all_aboard  \
5003    Eurocopter AS350B3 Ecureuil       N351SH    4598          6   
5004  Beechcraft B300 King Air 350i       NAF203  FL-891         11   
5005               Beechcraft 1900D         4610   E-325         14   
5006       Lockheed C-130H Hercules         5125    5125         96   
5007             Antonov An 26B-100     RA-26085   12310         28   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
5003                 5           1                      5   
5004                 7           4                     11   
5005                12           2                     12   
5006                88           8                     50   
5007                22           6                     28   

     passenger_fatalities crew_fatalities ground  \
5003                    4               1      0   
5004                    7               4      0   
5005                   11               1      0   
5006                    ?               ?      3   
5007                   22               6      0   

                                                summary  
5003  The sightseeing helicopter crashed after missi...  
5004  While on final approach, in poor weather condi...  
5005  The plane was carrying military personnel and ...  
5006  While attempting to land at Jolo Airport, the ...  
5007  The passenger plane crashed into the top of a ...

In [4]:
#Visualizamos informacion general del Data Frame (Cant. de filas, tipos de datos, etc)
aviones_db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5008 entries, 0 to 5007
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Unnamed: 0              5008 non-null   int64 
 1   fecha                   5008 non-null   object
 2   HORA declarada          5008 non-null   object
 3   Ruta                    5008 non-null   object
 4   OperadOR                5008 non-null   object
 5   flight_no               5008 non-null   object
 6   route                   5008 non-null   object
 7   ac_type                 5008 non-null   object
 8   registration            5008 non-null   object
 9   cn_ln                   5008 non-null   object
 10  all_aboard              5008 non-null   object
 11  PASAJEROS A BORDO       5008 non-null   object
 12  crew_aboard             5008 non-null   object
 13  cantidad de fallecidos  5008 non-null   object
 14  passenger_fatalities    5008 non-null   object
 15  crew

In [5]:
#Visualizamos nulos 
aviones_db.isnull().sum()

Unnamed: 0                0
fecha                     0
HORA declarada            0
Ruta                      0
OperadOR                  0
flight_no                 0
route                     0
ac_type                   0
registration              0
cn_ln                     0
all_aboard                0
PASAJEROS A BORDO         0
crew_aboard               0
cantidad de fallecidos    0
passenger_fatalities      0
crew_fatalities           0
ground                    0
summary                   0
dtype: int64

Luego de este primer paneo del Data Frame podemos sacar varias conclusiones. 

Primero, que la primer columna es un indice duplicado, por lo que no hace falta. Segundo, que hay una gran falta de datos al principio del DF que seguramente de debe a que son datos de asi casi 100 años y en ese entonces no habia las mismas regulaciones que ahora, pero al final del DF cuando los datos son mas recientes, la informacion esta mas completa. Ademas, podemos ver que no hay nulos, pero que hay gran cantidad de signos de preguntas, que en este caso pueden significar lo mismo, es decir, falta de datos. Por utilmo, los tipos de datos no son los correctos para todas las columnas, se deben corregir. 

Para sacar mas conclusiones, es necesario continuar analisando. Vamos a empezar por corregir lo que acabamos de marcar. 

Antes que nada debemos volver hacer el conteo de nulos por columna, es decir, de "?".

In [6]:
#Cambiamos los "?", por nulos
aviones_db = aviones_db.replace("?", np.nan)

In [7]:
#Visualizamos los nulos nuevamente
aviones_db.isnull().sum()

Unnamed: 0                   0
fecha                        0
HORA declarada            1504
Ruta                         5
OperadOR                    10
flight_no                 3682
route                      762
ac_type                     13
registration               272
cn_ln                      667
all_aboard                  17
PASAJEROS A BORDO          221
crew_aboard                219
cantidad de fallecidos       8
passenger_fatalities       235
crew_fatalities            235
ground                      44
summary                     59
dtype: int64

Vemos que de 5008 filas, mas de 3500 no tienen el numero de vuelo, por lo que no vale la pena conservar la columna. 

In [8]:
#Eliminamos la primer columna
aviones_db.drop("flight_no", axis=1, inplace=True)

Para las columnas numericas vamos a cambiar el "?" por 0. Para las columnas alfanumericas vamos a cambiar el "?" por "Sin Dato". 

In [9]:
#Vamos a cambiar el "?" por 0
aviones_db["all_aboard"] = aviones_db["all_aboard"].replace(np.nan, 0)
aviones_db["PASAJEROS A BORDO"] = aviones_db["PASAJEROS A BORDO"].replace(np.nan, 0)
aviones_db["crew_aboard"] = aviones_db["crew_aboard"].replace(np.nan, 0)
aviones_db["cantidad de fallecidos"] = aviones_db["cantidad de fallecidos"].replace(np.nan, 0)
aviones_db["passenger_fatalities"] = aviones_db["passenger_fatalities"].replace(np.nan, 0)
aviones_db["crew_fatalities"] = aviones_db["crew_fatalities"].replace(np.nan, 0)
aviones_db["ground"] = aviones_db["ground"].replace(np.nan, 0)

In [10]:
#Vamos a cambiar el "?" por "Sin Dato"
aviones_db["Ruta"] = aviones_db["Ruta"].replace(np.nan, "Sin Dato")
aviones_db["OperadOR"] = aviones_db["OperadOR"].replace(np.nan, "Sin Dato")
aviones_db["route"] = aviones_db["route"].replace(np.nan, "Sin Dato")
aviones_db["ac_type"] = aviones_db["ac_type"].replace(np.nan, "Sin Dato")
aviones_db["registration"] = aviones_db["registration"].replace(np.nan, "Sin Dato")
aviones_db["cn_ln"] = aviones_db["cn_ln"].replace(np.nan, "Sin Dato")
aviones_db["summary"] = aviones_db["summary"].replace(np.nan, "Sin Dato")

In [11]:
#Eliminamos la primer columna
aviones_db.drop("Unnamed: 0", axis=1, inplace=True)

In [12]:
#cambiamos la feacha a formato fecha
aviones_db["fecha"] = pd.to_datetime(aviones_db["fecha"])

In [13]:
#Cambiamos la hora a formato hora
aviones_db["HORA declarada"] = pd.to_datetime(aviones_db["HORA declarada"], format="%H%M", errors="coerce").dt.time

In [14]:
#Cambiamos las columnas numericas por tipo int
aviones_db["all_aboard"] = aviones_db["all_aboard"].astype(int)
aviones_db["all_aboard"] = aviones_db["all_aboard"].astype(int)
aviones_db["PASAJEROS A BORDO"] = aviones_db["PASAJEROS A BORDO"].astype(int)
aviones_db["crew_aboard"] = aviones_db["crew_aboard"].astype(int)
aviones_db["cantidad de fallecidos"] = aviones_db["cantidad de fallecidos"].astype(int)
aviones_db["passenger_fatalities"] = aviones_db["passenger_fatalities"].astype(int)
aviones_db["crew_fatalities"] = aviones_db["crew_fatalities"].astype(int)
aviones_db["ground"] = aviones_db["ground"].astype(int)

In [15]:
#Visualizamos el DF luego de las transformaciones
aviones_db.head()

fecha HORA declarada                                Ruta  \
0 1908-09-17       17:18:00                 Fort Myer, Virginia   
1 1909-09-07            NaT             Juvisy-sur-Orge, France   
2 1912-07-12       06:30:00           Atlantic City, New Jersey   
3 1913-08-06            NaT  Victoria, British Columbia, Canada   
4 1913-09-09       18:30:00                  Over the North Sea   

                 OperadOR          route                 ac_type registration  \
0    Military - U.S. Army  Demonstration        Wright Flyer III     Sin Dato   
1                Sin Dato       Air show          Wright Byplane          SC1   
2    Military - U.S. Navy    Test flight               Dirigible     Sin Dato   
3                 Private       Sin Dato        Curtiss seaplane     Sin Dato   
4  Military - German Navy       Sin Dato  Zeppelin L-1 (airship)     Sin Dato   

      cn_ln  all_aboard  PASAJEROS A BORDO  crew_aboard  \
0         1           2                  1            1   
1  Sin Dato           1                  0            1   
2  Sin Dato           5                  0            5   
3  Sin Dato           1                  0            1   
4  Sin Dato          20                  0            0   

   cantidad de fallecidos  passenger_fatalities  crew_fatalities  ground  \
0                       1                     1                0       0   
1                       1                     0                0       0   
2                       5                     0                5       0   
3                       1                     0                1       0   
4                      14                     0                0       0   

                                             summary  
0  During a demonstration flight, a U.S. Army fly...  
1  Eugene Lefebvre was the first pilot to ever be...  
2  First U.S. dirigible Akron exploded just offsh...  
3  The first fatal airplane accident in Canada oc...  
4  The airship flew into a thunderstorm and encou...

In [16]:
#Visualizamos informacion general
aviones_db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5008 entries, 0 to 5007
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   fecha                   5008 non-null   datetime64[ns]
 1   HORA declarada          3287 non-null   object        
 2   Ruta                    5008 non-null   object        
 3   OperadOR                5008 non-null   object        
 4   route                   5008 non-null   object        
 5   ac_type                 5008 non-null   object        
 6   registration            5008 non-null   object        
 7   cn_ln                   5008 non-null   object        
 8   all_aboard              5008 non-null   int64         
 9   PASAJEROS A BORDO       5008 non-null   int64         
 10  crew_aboard             5008 non-null   int64         
 11  cantidad de fallecidos  5008 non-null   int64         
 12  passenger_fatalities    5008 non-null   int64   

In [17]:
#Visualizamos nulos
aviones_db.isnull().sum()

fecha                        0
HORA declarada            1721
Ruta                         0
OperadOR                     0
route                        0
ac_type                      0
registration                 0
cn_ln                        0
all_aboard                   0
PASAJEROS A BORDO            0
crew_aboard                  0
cantidad de fallecidos       0
passenger_fatalities         0
crew_fatalities              0
ground                       0
summary                      0
dtype: int64

Vemos que hay una gran cantidad de nulos en la hora declarada. Por ahora no vamos hacer modificaciones.

In [18]:
#Buscamos filas duplicadas
filas_duplicadas = aviones_db.duplicated()
print(aviones_db[filas_duplicadas])

Empty DataFrame
Columns: [fecha, HORA declarada, Ruta, OperadOR, route, ac_type, registration, cn_ln, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, summary]
Index: []


Es un buena practica, observar los datos dentro de cada columna, en busca de anomalias. 

In [19]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["fecha"])

[1908-09-17, 1909-09-07, 1912-07-12, 1913-08-06, 1913-09-09, ..., 2021-03-28, 2021-05-21, 2021-06-10, 2021-07-04, 2021-07-06]
Length: 5008
Categories (4577, datetime64[ns]): [1908-09-17, 1909-09-07, 1912-07-12, 1913-08-06, ..., 2021-05-21, 2021-06-10, 2021-07-04, 2021-07-06]

In [20]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["HORA declarada"])

[17:18:00, NaN, 06:30:00, NaN, 18:30:00, ..., 18:35:00, 18:00:00, 08:00:00, NaN, 15:00:00]
Length: 5008
Categories (1036, object): [00:00:00, 00:01:00, 00:02:00, 00:04:00, ..., 23:56:00, 23:57:00, 23:58:00, 23:59:00]

In [21]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["Ruta"])

['Fort Myer, Virginia', 'Juvisy-sur-Orge, France', 'Atlantic City, New Jersey', 'Victoria, British Columbia, Canada', 'Over the North Sea', ..., 'Near Butte, Alaska', 'Near Kaduna, Nigeria', 'Near Pyin Oo Lwin, Myanmar', 'Patikul, Sulu, Philippines', 'Palana, Russia']
Length: 5008
Categories (4125, object): ['1,200 miles off Dakar, Atlantic Ocean', '110 miles SW of Sochi, Russia', '125 miles ENE of Tokyo, Japan', '175 miles off the Egyptian coast', ..., 'Zifta, Egypt', 'Zurich, Switzerland', 'Zwingle, Iowa', 'near Craigavad, Northern Ireland']

In [22]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["OperadOR"])

['Military - U.S. Army', 'Sin Dato', 'Military - U.S. Navy', 'Private', 'Military - German Navy', ..., 'Soloy Helicopters', 'Military - Nigerian Air Force', 'Military - Myanmar Air Force', 'Military - Philippine Air Force', 'Kamchatka Aviation Enterprise']
Length: 5008
Categories (2268, object): ['46826/109', 'A B Aerotransport', 'AB Aerotransport', 'ACES Colombia', ..., 'de Havilland Air Service', 'de Havilland Dove', 'u.s. Air Force', 'èkoda (India) Ltd']

In [23]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["route"])

['Demonstration', 'Air show', 'Test flight', 'Sin Dato', 'Sin Dato', ..., 'Sightseeing Charter', 'Sin Dato', 'Naypyidaw - Anisakan', 'Cagayan de Oro-Lumbia - Jolo', 'Petropavlovsk - Palana']
Length: 5008
Categories (3839, object): ['14099', ':  Comox-Port Hardy', 'Abadan - Tehran', 'Abakan - Petropavlosk - Kamchatsky', ..., 'bang - Bandar Lampung - Jakarta', 'on - Quang Ngai', 'test flight', 'thampton - Lisbon - Funchal']

In [24]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["ac_type"])

['Wright Flyer III', 'Wright Byplane', 'Dirigible', 'Curtiss seaplane', 'Zeppelin L-1 (airship)', ..., 'Eurocopter AS350B3 Ecureuil', 'Beechcraft B300 King Air 350i', 'Beechcraft 1900D', 'Lockheed C-130H Hercules', 'Antonov An 26B-100']
Length: 5008
Categories (2469, object): ['A-7D Corsair', 'AAC-1 Toucan', 'AEGK', 'AT L98 Carvair', ..., 'de Hvilland 89A Dragon Rapide', 'de havilland 89 Dragon Rapide', 'de havilland Canada Twin Otter 200', 'deHavilland DH-86']

In [25]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["registration"])

['Sin Dato', 'SC1', 'Sin Dato', 'Sin Dato', 'Sin Dato', ..., 'N351SH', 'NAF203', '4610', '5125', 'RA-26085']
Length: 5008
Categories (4701, object): ['00-0173', '011', '01710 / 69113', '019', ..., 'ZS-OYI', 'ZS-PDV', 'ZS-SAS', 'ZS-UAS']

In [26]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["cn_ln"])

['1', 'Sin Dato', 'Sin Dato', 'Sin Dato', 'Sin Dato', ..., '4598', 'FL-891', 'E-325', '5125', '12310']
Length: 5008
Categories (3908, object): ['0001', '0001A', '001', '0012', ..., 'W8-7', 'X0101', 'ZR726001', 'ZZX3201']

In [27]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["all_aboard"])

[2, 1, 5, 1, 20, ..., 6, 11, 14, 96, 28]
Length: 5008
Categories (244, int64): [0, 1, 2, 3, ..., 394, 517, 524, 644]

In [28]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["PASAJEROS A BORDO"])

[1, 0, 0, 0, 0, ..., 5, 7, 12, 88, 22]
Length: 5008
Categories (234, int64): [0, 1, 2, 3, ..., 381, 503, 509, 614]

In [29]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["crew_aboard"])

[1, 1, 5, 1, 0, ..., 1, 4, 2, 8, 6]
Length: 5008
Categories (34, int64): [0, 1, 2, 3, ..., 43, 49, 61, 83]

In [30]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["cantidad de fallecidos"])

[1, 1, 5, 1, 14, ..., 5, 11, 12, 50, 28]
Length: 5008
Categories (199, int64): [0, 1, 2, 3, ..., 346, 349, 520, 583]

In [31]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["passenger_fatalities"])

[1, 0, 0, 0, 0, ..., 4, 7, 11, 0, 22]
Length: 5008
Categories (190, int64): [0, 1, 2, 3, ..., 316, 335, 505, 560]

In [32]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["crew_fatalities"])

[0, 0, 5, 1, 0, ..., 1, 4, 1, 0, 6]
Length: 5008
Categories (28, int64): [0, 1, 2, 3, ..., 25, 27, 33, 43]

In [33]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["ground"])

[0, 0, 0, 0, 0, ..., 0, 0, 0, 3, 0]
Length: 5008
Categories (51, int64): [0, 1, 2, 3, ..., 113, 125, 225, 2750]

In [34]:
#Chequeamos los datos de cada columna
pd.Categorical(aviones_db["summary"])

['During a demonstration flight, a U.S. Army fl..., 'Eugene Lefebvre was the first pilot to ever b..., 'First U.S. dirigible Akron exploded just offs..., 'The first fatal airplane accident in Canada o..., 'The airship flew into a thunderstorm and enco..., ..., 'The sightseeing helicopter crashed after miss..., 'While on final approach, in poor weather cond..., 'The plane was carrying military personnel and..., 'While attempting to land at Jolo Airport, the..., 'The passenger plane crashed into the top of a...]
Length: 5008
Categories (4858, object): ['A 16 month old child was asphyxiated by a sea..., 'A Boeing 777-200 airliner operating as  Malay..., 'A Catalina seaplane was hijacked by a group a..., 'A Japan Air Self Defense Force F-86F fighter ..., ..., 'he United Nations-chartered plane was shot do..., 'the aircraft hit the slope of mountain locate..., 'the airplane hit the slope of the Cerro del C..., 'the crew failed to realize their altitude was...]

Cuando observamos cuantas categorias distintas de datos hay y como son algunas de ellas nos podemos dar una mejor idea de lo que contienen las columnas. 
En registracion veo una gran variabilidad entre las distintas categorias, por lo que van a ser dificiles de analizar. En cn_nl, pasa lo mismo que el caso anterior. Ademas, nos damos cuenta que la columna ruta y la columna route, a pesar de tener el mismo significado, dentro possen datos distintos. Por ultimo, nos damos cuenta que hay ciertos resumenes del hecho que estan repetidos, lo debemos tener en cuenta. 

Luego de una breve investigacion, averiguamos que la columna registracion se refiere un numero de identificacion que le da la IACO a ciertos aviones que cumplen con determiandos requisitos y que se comenzo a llevar a cabo hace menos de 100 años. En cuanto a cn_ln, se refiere al numero de serie del avion. 

Columna "summmary"
Vamos a analizar la cantidad de duplicados de esta columna, los registros que tengan el mismo resumen seran elimininados, ya que consideramos que es mas probable que se hayan duplicado los datos a que hayan sido dos accidentes identicos. 

In [35]:
#Localizamos todas las filas con el resumen repetido
filas_duplicadas = aviones_db[aviones_db.duplicated(subset=["summary"], keep=False)].index
print(filas_duplicadas)

Int64Index([  11,   16,   18,   25,   28,   58,   59,   74,   86,  103,
            ...
            4095, 4115, 4208, 4209, 4254, 4333, 4378, 4466, 4551, 4597],
           dtype='int64', length=195)


In [36]:
#Visualizamos cuantos de esos registros estan repetidos por ser "Sin Dato". Los restantes tienen el resumen repetido. 
frecuencia_sin_dato = aviones_db["summary"].value_counts()["Sin Dato"]
print(frecuencia_sin_dato)

59


In [37]:
#Visualizamos algun caso aleatorio para ver el comportamiento real
resumen_repetido = aviones_db.iloc[[86, 74]]
print(resumen_repetido)

        fecha HORA declarada              Ruta                OperadOR  \
86 1923-12-06            NaT    Castalia, Ohio  US Aerial Mail Service   
74 1922-10-25            NaT  Larache, Morocco  Grands Express Aeriens   

       route            ac_type registration     cn_ln  all_aboard  \
86  Sin Dato  De Havilland DH-4          318  Sin Dato           1   
74  Sin Dato         Breguet 14     Sin Dato  Sin Dato           2   

    PASAJEROS A BORDO  crew_aboard  cantidad de fallecidos  \
86                  0            1                       1   
74                  1            1                       2   

    passenger_fatalities  crew_fatalities  ground  \
86                     0                1       0   
74                     1                1       0   

                                 summary  
86  Crashed under unknown circumstances.  
74  Crashed under unknown circumstances.  


In [38]:
#Invetigamos otros casos repetidos
resumen_repetido = aviones_db.iloc[[16, 18]]
print(resumen_repetido)

        fecha HORA declarada                            Ruta  \
16 1917-06-14       08:45:00  Off Vlieland Island, North Sea   
18 1917-08-21       07:00:00             Off western Denmark   

                  OperadOR     route                  ac_type registration  \
16  Military - German Navy  Sin Dato  Zeppelin L-43 (airship)     Sin Dato   
18  Military - German Navy  Sin Dato  Zeppelin L-23 (airship)     Sin Dato   

       cn_ln  all_aboard  PASAJEROS A BORDO  crew_aboard  \
16  Sin Dato          24                  0            0   
18  Sin Dato          18                  0            0   

    cantidad de fallecidos  passenger_fatalities  crew_fatalities  ground  \
16                      24                     0                0       0   
18                      18                     0                0       0   

                           summary  
16  Shot down by British aircraft.  
18  Shot down by British aircraft.  


Luego de indagar sobre los resumenes repetidos, vemos que hay casos que ameritan el mismo resumen y son distintos. Por esta razon, no vamos a borrar ningun registro que tenga resumen repetido. 

Columna "cn_ln" y "regristration"

Vamos a ver si hay datos repetidos.

In [39]:
#Localizamos todas las filas con "cn_ln" repetido
duplicados_cn_ln = aviones_db[aviones_db.duplicated(subset=["cn_ln"], keep=False)].index
print(duplicados_cn_ln)

Int64Index([   0,    1,    2,    3,    4,    5,    6,    7,    8,    9,
            ...
            4952, 4957, 4959, 4961, 4964, 4967, 4974, 4987, 4989, 4996],
           dtype='int64', length=1400)


In [40]:
#Visualizamos cuantos de esos registros estan repetidos por ser "Sin Dato". Los restantes tienen el resumen repetido. 
frecuencia_sin_dato = aviones_db["cn_ln"].value_counts()["Sin Dato"]
print(frecuencia_sin_dato)

667


In [41]:
#Localizamos todas las filas con "registration" repetido
duplicados_cn_ln = aviones_db[aviones_db.duplicated(subset=["cn_ln"], keep=False)].index
print(duplicados_cn_ln)

Int64Index([   0,    1,    2,    3,    4,    5,    6,    7,    8,    9,
            ...
            4952, 4957, 4959, 4961, 4964, 4967, 4974, 4987, 4989, 4996],
           dtype='int64', length=1400)


In [42]:
#Visualizamos cuantos de esos registros estan repetidos por ser "Sin Dato". Los restantes tienen el resumen repetido. 
frecuencia_sin_dato = aviones_db["registration"].value_counts()["Sin Dato"]
print(frecuencia_sin_dato)

272


El uso que le vamos a dar a estas columnas es el dar la posibilidad posibilidad de buscar informacion extra del caso a la persona que este analisando el informe. Por lo que solo requerimos de una de ellas. Luego de investigar en internet, los datos de la columna "registration" nos dieron mejores resultados, sumado a que tiene menos registros sin dato. Por ello, borraremos la columna cn_ln.

In [43]:
#eliminamos la columna
aviones_db.drop(["cn_ln"], axis=1, inplace=True)

Al haber muchos tipos de aviones (militares, de carga, de pasajeros, etc) y que todos contienen distintos porcentajes de tribulantes y pasajeros, vamos a dejar de lado las columnas que lo especifican, quedandonos solo con el total de pasajeros abordo y de muertes en general. 

In [44]:
#Observamos los datos faltan en los totales
frecuencia_ceros = aviones_db["all_aboard"].value_counts().get(0, 0)
print(frecuencia_ceros)

22


In [45]:
#En esta columna no neceariamente son datos faltantes
frecuencia_ceros = aviones_db["cantidad de fallecidos"].value_counts().get(0, 0)
print(frecuencia_ceros)

84


Vamos a tratar de minimizar la perdida de datos, para ello en los registros que son igual a cero en los totales, vamos remplazar el cero por la suma de las otras dos columnas.

In [46]:
#Reemplazar los valores cero en la columna "all_aboard" por la suma de "PASAJEROS A BORDO" y "crew_aboard"
aviones_db.loc[aviones_db["all_aboard"] == 0, "all_aboard"] = aviones_db["PASAJEROS A BORDO"] + aviones_db["crew_aboard"]

In [47]:
#Reemplazar los valores cero en la columna "cantidad de fallecidos" por la suma de "passenger_fatalities" y "crew_fatalities"
aviones_db.loc[aviones_db["cantidad de fallecidos"] == 0, "cantidad de fallecidos"] = aviones_db["passenger_fatalities"] + aviones_db["crew_fatalities"]

In [48]:
#Observamos los datos faltan en los totales despues de las modificaciones
frecuencia_ceros = aviones_db["all_aboard"].value_counts().get(0, 0)
print(frecuencia_ceros)

19


In [49]:
#Observamos los datos faltan en los totales despues de las modificaciones
frecuencia_ceros = aviones_db["cantidad de fallecidos"].value_counts().get(0, 0)
print(frecuencia_ceros)

80


No se vieron grandes cambios, pero si hubo modificaciones. Ademas, era necesario corroborarlo. 
Ahora eliminamos las columnas que no utilizaremos.

In [50]:
#Eliminamos las columnas
aviones_db.drop(["PASAJEROS A BORDO", "crew_aboard", "passenger_fatalities", "crew_fatalities"], axis=1, inplace=True)

Visualizamos los registros sin pasajeros. 

In [51]:
sin_pasajeros = aviones_db.loc[aviones_db["all_aboard"] == 0]
sin_pasajeros.head(19)

fecha HORA declarada                                     Ruta  \
26   1919-07-21       16:55:00                        Chicago, Illinois   
103  1925-09-07            NaT                             Toul, France   
347  1934-08-10            NaT                            Ningbo, China   
384  1935-08-13            NaT                            Hangow, China   
451  1936-12-26            NaT                           Nanking, China   
652  1942-10-01            NaT                           Kunming, China   
754  1944-11-09            NaT                          Seljord, Norway   
854  1946-03-18            NaT    Between Chungking and Shanghai, China   
1231 1951-01-20            NaT                             Huilyo, Peru   
1362 1952-12-22            NaT                          Damascus, Syria   
1386 1953-04-10            NaT             Near Anchieta Island, Brazil   
1451 1954-03-13            NaT             Dien Bien Phu, South Vietnam   
1577 1956-06-13            NaT                      Over North Carolina   
1627 1957-04-20            NaT                            Jirkouk, Iraq   
1638 1957-06-24            NaT       Horn Island, Queensland, Australia   
3144 1981-09-22            NaT                    Near Babaeski, Turkey   
3376 1986-03-27            NaT         Bangui, Central African Republic   
3553 1989-05-09            NaT                      Near Tainjin, China   
3752 1992-02-20            NaT  En route from Argentina  to  California   

                                 OperadOR                            route  \
26     Wingfoot Air Express Goodyear Tire                          Shuttle   
103                                 CIDNA                         Sin Dato   
347   China National Aviation Corporation                         Sin Dato   
384   China National Aviation Corporation                         Sin Dato   
451   China National Aviation Corporation                         Sin Dato   
652   China National Aviation Corporation                         Sin Dato   
754        Military - U.S. Army Air Corps                         Sin Dato   
854   China National Aviation Corporation              Chunking - Shanghai   
1231                       Military - TAM                         Sin Dato   
1362                           Syrian Air                         Sin Dato   
1386                                 REAL                         Sin Dato   
1451                           Aigle Azur                         Sin Dato   
1577                             Piedmont                         Sin Dato   
1627                           Air France  Tehran, Iran - Istanbul, Turkey   
1638                      Adastra Airways                         Sin Dato   
3144         Military - Turkish Air Force                         Sin Dato   
3376          Military - French Air Force                         Sin Dato   
3553                           Aero Asahi                         Sin Dato   
3752                Aerolineas Argentinas                         Sin Dato   

                           ac_type registration  all_aboard  \
26               FD Type Dirigible     Sin Dato           0   
103                        Spad 33     Sin Dato           0   
347                 Sikorsky S-38B     Sin Dato           0   
384                 Sikorsky S-38B        NV40V           0   
451                   Douglas DC-2      NC14269           0   
652                   Douglas C-47           69           0   
754                       Sin Dato     42-52196           0   
854                       Sin Dato          139           0   
1231                  Douglas DC-3     Sin Dato           0   
1362                  Douglas DC-3     Sin Dato           0   
1386                  Doublas Dc-3     Sin Dato           0   
1451                          C-46     Sin Dato           0   
1577                  Douglas DC-3     Sin Dato           0   
1627  Lockheed Super Constellation       F-BGNE           0   
163

Observamos solamente 3 registros significativos, son los tres que tienen muertes de personas en tierra. Son los unicos que vamos a conservar.

In [52]:
#Obtenemos los índices de los registros con valor 0 en "all_aboard", excepto los índices 26, 3144 y 3376
indices_borrar = aviones_db[(aviones_db["all_aboard"] == 0) & (~aviones_db.index.isin([26, 3144, 3376]))].index

#Borramos los registros
aviones_db.drop(indices_borrar, inplace=True)

In [53]:
aviones_db.head()

fecha HORA declarada                                Ruta  \
0 1908-09-17       17:18:00                 Fort Myer, Virginia   
1 1909-09-07            NaT             Juvisy-sur-Orge, France   
2 1912-07-12       06:30:00           Atlantic City, New Jersey   
3 1913-08-06            NaT  Victoria, British Columbia, Canada   
4 1913-09-09       18:30:00                  Over the North Sea   

                 OperadOR          route                 ac_type registration  \
0    Military - U.S. Army  Demonstration        Wright Flyer III     Sin Dato   
1                Sin Dato       Air show          Wright Byplane          SC1   
2    Military - U.S. Navy    Test flight               Dirigible     Sin Dato   
3                 Private       Sin Dato        Curtiss seaplane     Sin Dato   
4  Military - German Navy       Sin Dato  Zeppelin L-1 (airship)     Sin Dato   

   all_aboard  cantidad de fallecidos  ground  \
0           2                       1       0   
1           1                       1       0   
2           5                       5       0   
3           1                       1       0   
4          20                      14       0   

                                             summary  
0  During a demonstration flight, a U.S. Army fly...  
1  Eugene Lefebvre was the first pilot to ever be...  
2  First U.S. dirigible Akron exploded just offsh...  
3  The first fatal airplane accident in Canada oc...  
4  The airship flew into a thunderstorm and encou...

Columna "operador".
A partir de esta columna, vamos a crear una nueva que nos indique sin es un avion militar o no.

In [54]:
#Visualizamos los distintos operadores
datos_operador = aviones_db["OperadOR"].unique().tolist()
print(datos_operador)

['Military - U.S. Army', 'Sin Dato', 'Military - U.S. Navy', 'Private', 'Military - German Navy', 'Military - German Army', 'US Aerial Mail Service', 'Wingfoot Air Express Goodyear Tire', 'Caproni Company', 'Aircraft Transport and Travel', 'Aircraft Travel Transport', 'Compañia Colombiana de Navegación Aérea', 'By Air', 'Latecoere Airlines', 'Handley Page Transport', 'Aeropostale', 'Military - Royal Australian Air Force', 'Military - Royal Airship Works', 'Franco-Roumaine', 'West Australian Airways', 'Military - U.S. Army Air Service', 'Aero Limited', 'Daimler Airways / Grands Express Aeriens', 'Cie des Messageries Aeriennes', 'Compagnie Franco-Roumaine de Navigaation Aerienne', 'Grands Express Aeriens', 'de Havilland Air Service', "Amee de l'Air", 'Grands Express Aeriens (Air Union)', 'Air Union', 'Campagnie France Roumaine', 'Daimler Airways', 'CCCP', 'Military - French Navy', 'KLM Royal Dutch Airlines', 'SCADTA', 'Imperial Airways', 'Zakavia', 'Lignes Aeriennes Latecoere', 'CIDNA', 

In [55]:
#Creamos la nueva columna 
aviones_db["militar"] = aviones_db["OperadOR"].str.contains(r"military|force", case=False).astype(int)

In [56]:
#Observamos cuantos aviones no militares hay
no_militar = aviones_db["militar"].value_counts().get(0, 0)
print(no_militar)

4202


Aproximadamente el 15% de los aviones son militares

In [57]:
#Observamos si hay una cantidad significativa de vuelos privados
private = aviones_db.loc[aviones_db["OperadOR"] == "Private"]
private.count()

fecha                     31
HORA declarada            22
Ruta                      31
OperadOR                  31
route                     31
ac_type                   31
registration              31
all_aboard                31
cantidad de fallecidos    31
ground                    31
summary                   31
militar                   31
dtype: int64

Luego de visualizar el archivo, llegamos a la conclusion de que la columna "ruta" es el lugar del accidente y que la columna "route" es el camino que recorre el avion. 
Vamos a analizar las dos columnas.

In [58]:
#Observamos nulos
ruta_nulos = aviones_db.loc[aviones_db["Ruta"] == "Sin Dato"]
ruta_nulos.count()

fecha                     5
HORA declarada            0
Ruta                      5
OperadOR                  5
route                     5
ac_type                   5
registration              5
all_aboard                5
cantidad de fallecidos    5
ground                    5
summary                   5
militar                   5
dtype: int64

In [59]:
#Observamos nulos
route_nulos = aviones_db.loc[aviones_db["route"] == "Sin Dato"]
ruta_nulos.count()

fecha                     5
HORA declarada            0
Ruta                      5
OperadOR                  5
route                     5
ac_type                   5
registration              5
all_aboard                5
cantidad de fallecidos    5
ground                    5
summary                   5
militar                   5
dtype: int64

Vamos a eliminar la columna "ruote" y ver que pasa en los 5 registros sin "Ruta"

In [60]:
#Eliminamos la columna
aviones_db.drop(["route"], axis=1, inplace=True)

In [61]:
#Observamos los registros sin dato en "Ruta"
sin_ruta = aviones_db.loc[aviones_db["Ruta"] == "Sin Dato"]
sin_ruta.head()

fecha HORA declarada      Ruta                             OperadOR  \
360  1934-12-31            NaT  Sin Dato                             Aeroflot   
465  1937-06-20            NaT  Sin Dato                                 ENTA   
646  1942-08-21            NaT  Sin Dato                   Deutsche Lufthansa   
999  1947-11-27            NaT  Sin Dato  China National Aviation Corporation   
2895 1977-01-20            NaT  Sin Dato                     North Canada Air   

                        ac_type registration  all_aboard  \
360               Tupolev ANT-9     Sin Dato          10   
465             Travel Air 6000         TI-3           6   
646               Siebel Si-204     Sin Dato           4   
999                Douglas DC-3     Sin Dato           3   
2895  Bristol 170 Freighter 31M       C-FWAD           2   

      cantidad de fallecidos  ground  \
360                       10       0   
465                        6       0   
646                        4       0   
999                        2       0   
2895                       1       0   

                                                summary  militar  
360                Crashed under unknown circumstances.        0  
465   Hit trees and crashed in an isolated area loca...        0  
646             Lufthansa chairman, Von Gablenz killed.        0  
999   The cargo plane was shot down by communist ant...        0  
2895  The cargo plane stalled nearly vertical and cr...        0

Observamos que les faltan varios datos y que son vuelos de mucha antiguedad, por lo que los vamos a eliminar.

In [62]:
# Eliminamos los registros de la columna "Ruta" que sean iguales a "Sin Dato"
aviones_db.drop(aviones_db[aviones_db["Ruta"] == "Sin Dato"].index, inplace=True)

Columna "HORA declarada".
Vamos a definir si fue nocturno o diurno. 
Desde las 07:00 hasta las 18:59 sera considerado diurno, desde las 19:00 a las 06:59 sera nocturno. 

In [63]:
#Creamos la nueva columna con el horario
aviones_db["horario"] = np.where(
    aviones_db["HORA declarada"].isna(),  # Verificar si el valor es nulo
    "Sin Dato",  # Valor a asignar si es nulo
    np.where(
        aviones_db["HORA declarada"].between(pd.to_datetime("07:00:00").time(), pd.to_datetime("18:59:00").time()),
        "Diurno",
        np.where(
            aviones_db["HORA declarada"].between(pd.to_datetime("19:00:00").time(), pd.to_datetime("23:59:59").time()) |
            aviones_db["HORA declarada"].between(pd.to_datetime("00:00:00").time(), pd.to_datetime("06:59:00").time()),
            "Nocturno",
            ""
        )
    )
)

In [64]:
#Eliminamos la columna hora
aviones_db.drop(["HORA declarada"], axis=1, inplace=True)

In [65]:
aviones_db.head()

fecha                                Ruta                OperadOR  \
0 1908-09-17                 Fort Myer, Virginia    Military - U.S. Army   
1 1909-09-07             Juvisy-sur-Orge, France                Sin Dato   
2 1912-07-12           Atlantic City, New Jersey    Military - U.S. Navy   
3 1913-08-06  Victoria, British Columbia, Canada                 Private   
4 1913-09-09                  Over the North Sea  Military - German Navy   

                  ac_type registration  all_aboard  cantidad de fallecidos  \
0        Wright Flyer III     Sin Dato           2                       1   
1          Wright Byplane          SC1           1                       1   
2               Dirigible     Sin Dato           5                       5   
3        Curtiss seaplane     Sin Dato           1                       1   
4  Zeppelin L-1 (airship)     Sin Dato          20                      14   

   ground                                            summary  militar  \
0       0  During a demonstration flight, a U.S. Army fly...        1   
1       0  Eugene Lefebvre was the first pilot to ever be...        0   
2       0  First U.S. dirigible Akron exploded just offsh...        1   
3       0  The first fatal airplane accident in Canada oc...        0   
4       0  The airship flew into a thunderstorm and encou...        1   

    horario  
0    Diurno  
1  Sin Dato  
2  Nocturno  
3  Sin Dato  
4    Diurno

Columna "fecha".
Le extraeremos el año y el mes, para luego eliminarla. 

In [66]:
#Creamo la nueva columna "año" con el año de la fecha 
aviones_db["año"] = pd.to_datetime(aviones_db["fecha"]).dt.year

In [67]:
#Creamo la nueva columna "mes" con el mes de la fecha 
aviones_db["mes"] = pd.to_datetime(aviones_db["fecha"]).dt.month_name()

In [68]:
#Creamo la nueva columna "dia" con el mes de la fecha 
aviones_db["dia"] = pd.to_datetime(aviones_db["fecha"]).dt.day_name()

In [69]:
#Eliminamos la columna fecha
aviones_db.drop(["fecha"], axis=1, inplace=True)

Columna "Ruta"

In [72]:
#Extraemos el país de la columna "Ruta"
def obtener_pais(row):
    partes = row["Ruta"].split(", ")
    if len(partes) == 1:
        return partes[0]
    elif len(partes) == 2:
        return partes[1]
    elif len(partes) >= 3:
        return partes[2]

#Aplicamos la función a la columna "Ruta" y asignamos los resultados a una nueva columna "Pais"
aviones_db["Pais"] = aviones_db.apply(obtener_pais, axis=1)

In [162]:
#Tratamos el caso especial de Estados Unidos
reemplazos = {
    'Alaska': 'United States',
    'Alaksa': 'United States',
    'Alabama': 'United States',
    'Arkansas': 'United States',
    'Arizona': 'United States',
    'California': 'United States',
    'Californiia': 'United States',
    'Cailifornia': 'United States',
    'Colorado': 'United States',
    'Columbia': 'United States',
    'Connecticut': 'United States',
    'Delaware': 'United States',
    'Florida': 'United States',
    'Georgia': 'United States',
    'Hawaii': 'United States',
    'Idaho': 'United States',
    'Illinois': 'United States',
    'Ilinois': 'United States',
    'Indiana': 'United States',
    'Iowa': 'United States',
    'Kansas': 'United States',
    'Kentucky': 'United States',
    'Louisiana': 'United States',
    'Maine': 'United States',
    'Maryland': 'United States',
    'Massachusetts': 'United States',
    'Michigan': 'United States',
    'Minnesota': 'United States',
    'Minnisota': 'United States',
    'Mississipi': 'United States',
    'Missouri': 'United States',
    'Montana': 'United States',
    'Nebraska': 'United States',
    'Nevada': 'United States',
    'New Hampshire': 'United States',
    'New Jersey': 'United States',
    'Jersey': 'United States',
    'New Mexico': 'United States',
    'New York': 'United States',
    'NY': 'United States',
    'North Carolina': 'United States',
    'North Dakota': 'United States',
    'Ohio': 'United States',
    'Oklahoma': 'United States',
    'Oregon': 'United States',
    'Pennsylvania': 'United States',
    'Rhode Island': 'United States',
    'South Carolina': 'United States',
    'South Dakota': 'United States',
    'Tennessee': 'United States',
    'Texas': 'United States',
    'Utah': 'United States',
    'Vermont': 'United States',
    'Virginia': 'United States',
    'Washington': 'United States',
    'West Virginia': 'United States',
    'Wisconsin': 'United States',
    'Wyoming': 'United States',
    'WY': 'United States'
}

aviones_db['Pais'] = aviones_db['Pais'].replace(reemplazos)

In [141]:
#Tratamos el caso especial de inglaterra
aviones_db['Pais'] = aviones_db['Pais'].replace('England', 'UK')
aviones_db.loc[aviones_db['Pais'].str.contains('United Kingdom', case=False), 'Pais'] = 'UK'
aviones_db.loc[aviones_db['Pais'].str.contains('Unied Kingdom', case=False), 'Pais'] = 'UK'
aviones_db.loc[aviones_db['Pais'].str.contains('Scotland', case=False), 'Pais'] = 'UK'

In [97]:
#Tratamos el caso especial de Ontario
aviones_db['Pais'] = aviones_db['Pais'].replace('Ontario', 'Canada')

In [108]:
#Caso Brazil
aviones_db.loc[aviones_db['Pais'].str.contains('Brazil', case=False), 'Pais'] = 'Brazil'

In [110]:
#Caso Francia
aviones_db.loc[aviones_db['Pais'].str.contains('France', case=False), 'Pais'] = 'France'

In [114]:
#Caso Francia
aviones_db.loc[aviones_db['Pais'].str.contains('Spain', case=False), 'Pais'] = 'Spain'

In [115]:
#Caso Francia
aviones_db.loc[aviones_db['Pais'].str.contains('Germany', case=False), 'Pais'] = 'Germany'

In [137]:
#Caso Malta
aviones_db.loc[aviones_db['Pais'].str.contains('Malta', case=False), 'Pais'] = 'Malta'

In [121]:
#Caso Holanda
aviones_db.loc[aviones_db['Pais'].str.contains('Netherlands', case=False), 'Pais'] = 'Netherlands'
aviones_db.loc[aviones_db['Pais'].str.contains('Amsterdam', case=False), 'Pais'] = 'Netherlands'

In [123]:
#Caso Morocco
aviones_db.loc[aviones_db['Pais'].str.contains('Morocco', case=False), 'Pais'] = 'Morocco'

In [145]:
#Caso Croatia
aviones_db.loc[aviones_db['Pais'].str.contains('Croatia', case=False), 'Pais'] = 'Croatia'
aviones_db.loc[aviones_db['Pais'].str.contains('Coatia', case=False), 'Pais'] = 'Croatia'

In [133]:
#Caso Portugal
aviones_db.loc[aviones_db['Pais'].str.contains('Lisbon', case=False), 'Pais'] = 'Portugal'

In [149]:
#Caso Australia
aviones_db.loc[aviones_db['Pais'].str.contains('Australia', case=False), 'Pais'] = 'Australia'
aviones_db.loc[aviones_db['Pais'].str.contains('Australila', case=False), 'Pais'] = 'Australia'

In [151]:
#Caso Australia
aviones_db.loc[aviones_db['Pais'].str.contains('Sweden', case=False), 'Pais'] = 'Sweden'
aviones_db.loc[aviones_db['Pais'].str.contains('Swden', case=False), 'Pais'] = 'Sweden'

In [160]:
#Caso Australia
aviones_db.loc[aviones_db['Pais'].str.contains('Bulgaria', case=False), 'Pais'] = 'Bulgaria'
aviones_db.loc[aviones_db['Pais'].str.contains('bulgaria', case=False), 'Pais'] = 'Bulgaria'
aviones_db.loc[aviones_db['Pais'].str.contains('bulgeria', case=False), 'Pais'] = 'Bulgaria'

In [155]:
#Caso Russia
aviones_db.loc[aviones_db['Pais'].str.contains('Rusia', case=False), 'Pais'] = 'Rusia'
aviones_db.loc[aviones_db['Pais'].str.contains('USSR', case=False), 'Pais'] = 'Rusia'
aviones_db.loc[aviones_db['Pais'].str.contains('Yugoslavia', case=False), 'Pais'] = 'Rusia'

In [181]:
#Caso Dinamarca
aviones_db.loc[aviones_db['Pais'].str.contains('Denmark', case=False), 'Pais'] = 'Denmark'

In [175]:
#Caso Ma del Norte
aviones_db.loc[aviones_db['Pais'].str.contains('North Sea', case=False), 'Pais'] = 'North Sea'

In [182]:
#Visualizamos los distintos paises
distintos_paises = aviones_db['Pais'].unique()
print(distintos_paises)

['United States' 'France' 'Canada' 'North Sea' 'Germany' 'Belgium'
 'Bulgaria' 'UK' 'Denmark' 'Over the Mediterranean' 'Italy' 'Kent' 'Spain'
 'Off Gibraltar' 'Australia' 'North Atlantic Ocean' 'China' 'Morocco'
 'Romania' 'Russia' 'Over the Mediterranean Sea'
 'Over the English Channel' 'Colombia' 'Rusia' 'Hungary' 'Mauritania'
 'English Channel' 'Czechoslovakia' 'Algeria' 'Atlantic Ocean'
 'Over the Gulf of Finland' 'Uruguay' 'Ariège' 'Brazil' 'Netherlands'
 'Greece' 'Mexico' ' Pest Hungary' 'Algiers' 'Cuba' 'Iran' 'East Sardinia'
 'Madagascar' 'Argentina' 'New Zealand' 'Burma' 'South Africa' 'Thailand'
 'Off Algiers' 'Iraq' 'Chile' 'Lebanon' 'Bolivia' 'Aegean Sea' 'Peru'
 'Syria' 'Pakistan' 'Egypt' 'Honduras' 'Switzerland' 'Panama'
 'Gulf of Finland' 'Poland' 'Boliva' 'South Atlantic Ocean' 'Trinidad'
 'Sweden' 'Norway' 'Croatia' 'Tajikistan' 'Ciudad Serdan,Mexico' 'Libya'
 'French West Africa' 'Gambia' 'Desertores Island Región de Los Lagos'
 'New Guinea' 'Sudan' 'Papua New Guinea'

In [105]:
#Eliminamos la columna "Ruta"
aviones_db.drop(["Ruta"], axis=1, inplace=True)

In [169]:
#Eliminamos la columna "Malta" que en algun momento creamos por error
aviones_db.drop(["Malta"], axis=1, inplace=True)

In [183]:
# Crear la nueva columna "mar" con valores 1 o 0
aviones_db['mar'] = aviones_db['Pais'].str.contains('Sea|Ocean|Channel|Gulf|Off', case=False, na=False).astype(int)

In [185]:
#Visualizacion final
aviones_db.head(5)

OperadOR                 ac_type registration  all_aboard  \
0    Military - U.S. Army        Wright Flyer III     Sin Dato           2   
1                Sin Dato          Wright Byplane          SC1           1   
2    Military - U.S. Navy               Dirigible     Sin Dato           5   
3                 Private        Curtiss seaplane     Sin Dato           1   
4  Military - German Navy  Zeppelin L-1 (airship)     Sin Dato          20   

   cantidad de fallecidos  ground  \
0                       1       0   
1                       1       0   
2                       5       0   
3                       1       0   
4                      14       0   

                                             summary  militar   horario   año  \
0  During a demonstration flight, a U.S. Army fly...        1    Diurno  1908   
1  Eugene Lefebvre was the first pilot to ever be...        0  Sin Dato  1909   
2  First U.S. dirigible Akron exploded just offsh...        1  Nocturno  1912   
3  The first fatal airplane accident in Canada oc...        0  Sin Dato  1913   
4  The airship flew into a thunderstorm and encou...        1    Diurno  1913   

         mes        dia           Pais  mar  
0  September   Thursday  United States    0  
1  September    Tuesday         France    0  
2       July     Friday  United States    0  
3     August  Wednesday         Canada    0  
4  September    Tuesday      North Sea    1

In [186]:
#Visualizacion inal
aviones_db.tail(5)

OperadOR                        ac_type  \
5003                Soloy Helicopters    Eurocopter AS350B3 Ecureuil   
5004    Military - Nigerian Air Force  Beechcraft B300 King Air 350i   
5005     Military - Myanmar Air Force               Beechcraft 1900D   
5006  Military - Philippine Air Force       Lockheed C-130H Hercules   
5007    Kamchatka Aviation Enterprise             Antonov An 26B-100   

     registration  all_aboard  cantidad de fallecidos  ground  \
5003       N351SH           6                       5       0   
5004       NAF203          11                      11       0   
5005         4610          14                      12       0   
5006         5125          96                      50       3   
5007     RA-26085          28                      28       0   

                                                summary  militar   horario  \
5003  The sightseeing helicopter crashed after missi...        0    Diurno   
5004  While on final approach, in poor weather condi...        1    Diurno   
5005  The plane was carrying military personnel and ...        1    Diurno   
5006  While attempting to land at Jolo Airport, the ...        1  Sin Dato   
5007  The passenger plane crashed into the top of a ...        0    Diurno   

       año    mes       dia           Pais  mar  
5003  2021  March    Sunday  United States    0  
5004  2021    May    Friday        Nigeria    0  
5005  2021   June  Thursday        Myanmar    0  
5006  2021   July    Sunday    Philippines    0  
5007  2021   July   Tuesday         Russia    0

In [188]:
#Exportamos
aviones_db.to_csv('/Users/tobiasolivapinto/Desktop/Proyecto Individual 2 - Henry/AccidentesAviones_ETL.csv', index=False)